src/protocol/util/websocket_server.cpp

```diff
@@ -0,0 +1,541 @@
+
+  public:
+    websocket_monitor_session(tcp::socket socket)
+        : ws_(std::move(socket))
+    {
+        auto endpoint = ws_.next_layer().remote_endpoint();
+        address_      = endpoint.address().to_string() + ":" + std::to_string(endpoint.port());
+    }
+
+    void start()
+    {
+        // Enable permessage-deflate compression for monitor sessions as well
+        websocket::permessage_deflate pmd;
+        pmd.server_enable = true;
+        pmd.client_enable = true;
+        ws_.set_option(pmd);
+
+        // Accept the websocket handshake
+        ws_.async_accept([self = shared_from_this()](beast::error_code ec) {
+            if (!ec) {
+                CASPAR_LOG(info) << L"WebSocket Monitor client connected: " << u16(self->address_);
+                self->do_read();
+            } else {
+                CASPAR_LOG(error) << L"WebSocket Monitor accept failed: " << u16(ec.message());
+            }
+        });
+    }
+
+    void send_monitor_data(const std::string& json_data)
+    {
+        auto self = shared_from_this();
+        net::post(ws_.get_executor(), [self, json_data]() {
+            try {
+                if (self->ws_.is_open()) {
+                    self->ws_.write(net::buffer(json_data));
+                }
+            } catch (const std::exception& e) {
+                CASPAR_LOG(error) << L"Failed to send monitor data to WebSocket client: " << u16(e.what());
+            }
+        });
+    }
+
+    bool is_open() const { return ws_.is_open(); }
+
+    std::string address() const { return address_; }
+
+  private:
+    void do_read()
+    {
+        auto self = shared_from_this();
+        // Monitor connections are read-only, but we still need to handle close events
+        ws_.async_read(buffer_, [self](beast::error_code ec, std::size_t) {
+            if (ec == websocket::error::closed) {
+                CASPAR_LOG(info) << L"WebSocket Monitor client disconnected: " << u16(self->address_);
+            } else if (ec) {
+                CASPAR_LOG(error) << L"WebSocket Monitor read error: " << u16(ec.message());
+            } else {
+                // Ignore incoming messages from monitor clients
+                self->buffer_.clear();
+                self->do_read();
+            }
+        });
+    }
+
+    beast::flat_buffer buffer_;
+};
+
+struct websocket_server::impl : public spl::enable_shared_from_this<websocket_server::impl>
+{
+    std::shared_ptr<boost::asio::io_context> context_;
+    protocol_strategy_factory<wchar_t>::ptr  amcp_protocol_factory_;
+    uint16_t                                 amcp_port_;
+    uint16_t                                 monitor_port_;
+
+    tcp::acceptor amcp_acceptor_;
+    tcp::acceptor monitor_acceptor_;
+
+    std::set<spl::shared_ptr<websocket_monitor_session>>                                           monitor_sessions_;
+    std::vector<std::function<std::pair<std::wstring, std::shared_ptr<void>>(const std::string&)>> lifecycle_factories_;
```

Comment:
as I said to some of this in server.cpp, I think a lot of this lifecycle logic can be removed